src/java.base/share/classes/com/sun/crypto/provider/Poly1305.java

```diff
@@ -255,6 +254,23 @@ private void processMultipleBlocks(byte[] input, int offset, int length, long[]
         }
     }
 
+    private void processMultipleBlocks(ByteBuffer buf, int blockMultipleLength) {
+        if (buf.hasArray()) {
+            byte[] input = buf.array();
+            int offset = buf.arrayOffset();
+            long[] aLimbs = a.getLimbs();
+            long[] rLimbs = r.getLimbs();
+
+            processMultipleBlocksCheck(input, offset, blockMultipleLength, aLimbs, rLimbs);
+            processMultipleBlocks(input, offset, blockMultipleLength, aLimbs, rLimbs);
+        } else {
+            while (blockMultipleLength > 0) {
+                processBlock(buf, BLOCK_LENGTH);
```

Comment:
For native byte buffers, you can pass the buffer address and base.

Reply:
Spent some time thinking about this one.. Originally, (from the sample you shared) I thought I had to use `Unsafe`, which I don't think is 'safe' to do in a crypto library. After some searching, I found a 'safe' alternative, `sun.nio.ch.DirectBuffer` interface, which would give me the `long address()`.

The problem is that the current java signature for the intrinsic expects a `[]byte` not a long. I could certainly change the existing intrinsic (or more likely add a new instrinsic calling existing stub assembly).

I would lean towards not supporting Direct Byte buffers right now. (Also, it doesn't seem like other crypto intrinsics do either. At least GCM does not..)

Reply:
JMH also added (and main PR comment modified to match)